kernel/src/mem.rs

```diff
@@ -69,213 +90,205 @@ unsafe fn raw_appslice_to_slice<'a>(ptr: *const u8, len: usize) -> &'a [u8] {
 /// within a single process' address space for the entire lifetime
 /// `'a`. No other mutable or immutable slice pointing to an
 /// overlapping memory region may exist over the entire lifetime `'a`.
```

Comment:
This is no longer true, correct?

Reply:
I believe Brad is correct. It is still true that no mutable slice (i.e. `&mut [u8]`) may exist pointing to a portion of the memory region, but an immutable slice pointing to that region should be fine?

Reply:
Yes, this comment is outdated, as it was intended to say that no other mutable or immutable _AppSlice_ (now ProcessBuffer) must exist while this is in scope. This is not the case any longer.

However, viewed from @hudson-ayers's perspective (when talking about Rust primitive slices of `u8`), this is still correct, both for immutable and mutable slices. This is the case for all process memory. When another immutable or mutable slice over said memory region is in scope, both the kernel changing memory through a slice of `Cell`s, as well as the process changing this memory will lead to unsoundness in the kernel.

I'll update the comment to incorporate these required invariants more precisely.